lib/HeroBonus.h

```diff
@@ -390,6 +407,7 @@ struct DLL_LINKAGE Bonus : public std::enable_shared_from_this<Bonus>
 #define BONUS_SOURCE(x) x,
 		BONUS_SOURCE_LIST
 #undef BONUS_SOURCE
+		MAX /*This is a dummy value, which will be always last*/

```

Comment:
Maybe BONUS_SOURCE_MAX?
This is not enum class so "MAX" will leak to entire Bonus class, which includes other enum's

Reply:
Renamed to NUM_BONUS_SOURCE.